File: src/main/java/br/com/dusty/dkits/listener/mechanics/ServerListPingListener.java
package br.com.dusty.dkits.listener.mechanics;

import br.com.dusty.dkits.Main;
import br.com.dusty.dkits.util.MOTDUtils;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.server.ServerListPingEvent;

public class ServerListPingListener implements Listener {
	
	@EventHandler
	public void onServerListPing(ServerListPingEvent event) {
		//TODO: Remove mods/admins from count
		event.setMaxPlayers(Main.MAX_PLAYERS);
		
		switch(Main.serverStatus){
			case ONLINE:
				event.setMotd(MOTDUtils.randomMOTD());
				break;
			case OFFLINE:
				event.setMotd(MOTDUtils.offlineMOTD());
				break;
			case MAINTENANCE:
				event.setMotd(MOTDUtils.maintenanceMOTD());
				break;
		}
	}
}

File: src/main/kotlin/br/com/dusty/dkits/listener/mechanics/ServerListPingListener.kt
package br.com.dusty.dkits.listener.mechanics

import br.com.dusty.dkits.Config
import br.com.dusty.dkits.gamer.EnumMode
import br.com.dusty.dkits.util.entity.gamer
import br.com.dusty.dkits.util.text.MOTDs
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.server.ServerListPingEvent

object ServerListPingListener: Listener {

	@EventHandler
	fun onServerListPing(event: ServerListPingEvent) {
		event.removeAll { event.iterator().next().gamer().mode == EnumMode.ADMIN }
		event.maxPlayers = Config.data.slots

		when (Config.data.serverStatus) {
			Config.EnumServerStatus.ONLINE      -> event.motd = MOTDs.randomMOTD()
			Config.EnumServerStatus.OFFLINE     -> event.motd = MOTDs.offlineMOTD()
			Config.EnumServerStatus.MAINTENANCE -> event.motd = MOTDs.maintenanceMOTD()
		}
	}
}
